pyartifactory/objects.py

```diff
@@ -1,684 +1,745 @@
+        self.get(username)
+        self._post(f"api/{self._uri}/{username}", json=user.dict())
+        logging.debug(f"User {username} successfully updated")
+        return self.get(username)
+
+    def delete(self, name: str) -> None:
+        """
+        Remove user
+        :param name: Name of the user to delete
+        :return: None
+        """
+        self.get(name)
+        self._delete(f"api/{self._uri}/{name}")
+        logging.debug(f"User {name} successfully deleted")
+
+
+class ArtifactorySecurity(ArtifactoryObject):
+    _uri = "security"
+
+    def __init__(self, artifactory: AuthModel) -> None:
+        super(ArtifactorySecurity, self).__init__(artifactory)
+
+    def get_encrypted_password(self) -> PasswordModel:
+        """
+        Get the encrypted password of the authenticated requestor.
+        :return: str
+        """
+        response = self._get(f"api/{self._uri}/encryptedPassword")
+        logging.debug("Encrypted password successfully delivered")
+        return PasswordModel(**response.json())
+
+    def create_access_token(
+        self,
+        user_name: str,
+        expires_in: int = 3600,
+        refreshable: bool = False,
+        groups: list = None,
+    ) -> AccessTokenModel:
+        """
+        Creates an access token.
+
+        :param user_name: Name of the user to whom an access key should be granted. transient token created if not set.
+        :param expires_in: Expiry time for the token in seconds. For eternal tokens specify 0.
+        :param refreshable: If set to true token can be refreshed using the refresh token returned. defaults False.
+        :param groups: A list of groups the token has membership of. If username is specified, groups are implied.
+        :return: AccessToken
+        """
+        payload = {
+            "username": user_name,
+            "expires_in": expires_in,
+            "refreshable": refreshable,
+        }
+        if groups:
+            if not isinstance(groups, list):
+                raise ValueError(groups)
+            scope = f'member-of-groups:"{", ".join(groups)}"'
+            payload.update({"scope": scope})
+        response = self._post(
+            f"api/{self._uri}/token", data=payload, raise_for_status=False
+        )
+        if response.ok:
+            return AccessTokenModel(**response.json())
+        raise InvalidTokenDataException(
+            response.json().get("error_description", "Unknown error")
+        )
+
+    def revoke_access_token(self, token: str = None, token_id: str = None) -> bool:
+        """
+        Revokes an access token.
+
+        :param token: The token to revoke
+        :param token_id: The id of a token to revoke
+        :return: bool True or False indicating success or failure of token revocation attempt.
+        """
+        if not any([token, token_id]):
+            logging.error("Neither a token or a token id was specified")
+            raise InvalidTokenDataException
```

Comment:
Could you please add a message a comment to the exception. For example, InvalidTokenDataException("Neither a token or a token id was specified")